app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -210,14 +209,21 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
             true;
         })
         mapController?.setFeatureTouchListener({
-            properties ->
-                val tapProp = properties.getString(MAP_DATA_PROP_TAP)
-                val searchIndexProp = properties.getNumber(MAP_DATA_PROP_SEARCHINDEX).toInt()
-                if (tapProp == MAP_DATA_PROP_TAP_SEARCH) {
-                    presenter?.onSearchResultTapped(searchIndexProp)
+            properties, positionX, positionY ->
+                // Reassign tapPoint to center of the feature tapped
+                // Also used in placing the pin
+                poiTapPoint = floatArrayOf(positionX, positionY)
+                if (properties.contains(MAP_DATA_PROP_SEARCHINDEX)) {
+                    val searchIndex = properties.getNumber(MAP_DATA_PROP_SEARCHINDEX).toInt()
+                    presenter?.onSearchResultTapped(searchIndex)
                 } else {
-                    if (poiTapPoint != null) {
-                        presenter?.onReverseGeoRequested(poiTapPoint?.get(0) as Float, poiTapPoint?.get(1) as Float)
+                    if (properties.contains(MAP_DATA_PROP_ID)) {
+                        val featureID = properties.getNumber(MAP_DATA_PROP_ID).toLong()
+                        presenter?.onPlaceSearchRequested("osm:venue:$featureID")
+                    } else {
+                        if (poiTapPoint != null) {
+                            presenter?.onReverseGeoRequested(poiTapPoint?.get(0) as Float, poiTapPoint?.get(1) as Float)
```

Comment:
It was failing to compile, because its expecting a `Float` and it found a `Float?`, so I added that `as`. Good to know the usage though.

Do you suggest another way to doing this? Checking if `poiTapPoint?.get(0)` is not null and they grabbing value from there?